cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -721,11 +721,12 @@ public boolean apply(RunRecordMeta invalidRunRecordMeta) {
       String runId = invalidRunRecordMeta.getPid();
       ProgramId targetProgramId = runIdToProgramId.get(runId);
 
-      LOG.warn("Fixing RunRecord {} in program {} of type {} with RUNNING status but the program is not running",
-               runId, targetProgramId, programType.getPrettyName());
-
-      store.compareAndSetStatus(targetProgramId, runId, ProgramController.State.ALIVE.getRunStatus(),
-                                ProgramController.State.ERROR.getRunStatus());
+      boolean updated = store.compareAndSetStatus(targetProgramId, runId, ProgramController.State.ALIVE.getRunStatus(),
+                                                  ProgramController.State.ERROR.getRunStatus());
+      if (updated) {
+        LOG.warn("Fixing RunRecord {} in program {} of type {} with RUNNING status but the program is not running",
```

Comment:
"Fixed RunRecord {} for program {} of type {} with RUNNING status, because the program is not actually running"?